postgres/datadog_checks/postgres/util.py

```diff
@@ -172,21 +172,28 @@ def get_schema_field(descriptors):
 REPLICATION_STATS_METRICS = {
     'descriptors': [('application_name', 'wal_app_name'), ('state', 'wal_state'), ('sync_state', 'wal_sync_state')],
     'metrics': {
-        'GREATEST (0, EXTRACT(epoch from write_lag)) as write_lag': (
+        'max(GREATEST (0, EXTRACT(epoch from write_lag))) as write_lag': (
             'postgresql.replication.wal_write_lag',
             AgentCheck.gauge,
         ),
-        'GREATEST (0, EXTRACT(epoch from flush_lag)) AS flush_lag': (
+        'max(GREATEST (0, EXTRACT(epoch from flush_lag))) AS flush_lag': (
             'postgresql.replication.wal_flush_lag',
             AgentCheck.gauge,
         ),
-        'GREATEST (0, EXTRACT(epoch from replay_lag)) AS replay_lag': (
+        'max(GREATEST (0, EXTRACT(epoch from replay_lag))) AS replay_lag': (
             'postgresql.replication.wal_replay_lag',
             AgentCheck.gauge,
         ),
+        'max(GREATEST (0, age(backend_xmin))) as backend_xmin_age': (
```

Comment:
If the type of `backend_xmin` is `xid` ([source](https://www.postgresql.org/docs/current/monitoring-stats.html#MONITORING-PG-STAT-REPLICATION-VIEW)), what is the behavior of `age(backend_xmin)` ? Won't the value be a nonsense number if age [expects a timestamp](https://www.postgresql.org/docs/current/functions-datetime.html)?

Reply:
actually, nevermind.  It requires `track_commit_timestamp` to be enabled, which is disabled by default.

Reply:
It looks like `age(xid)` is correct but undocumented. https://github.com/postgres/postgres/blob/REL_11_2/src/backend/utils/adt/xid.c#L103-L112

Reply:
Yeah, `age(xid)` gives the delta between latest stable xid on the primary and the provided value. This allows to not having to care about xid warparound.